test/gtest/ucs/test_pgtable.cc

```diff
@@ -524,7 +524,9 @@ class test_pgtable_perf : public test_pgtable {
 
 private:
     struct region_comparator {
-        bool operator()(ucs_pgt_region_t* region1, ucs_pgt_region_t* region2) {
+        bool
+        operator()(ucs_pgt_region_t *region1, ucs_pgt_region_t *region2) const
```

Comment:
does it also produce failures? (can't open a bug report)

Reply:
yes, when compiled UCX with gcc11 built from git, got this error:
```
  CXX      ucs/gtest-test_pgtable.o
In file included from $HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/set:60,
                 from $HOME/workspace/ucx/contrib/../test/gtest/common/gtest.h:3073,
                 from $HOME/workspace/ucx/contrib/../test/gtest/common/test_helpers.h:11,
                 from $HOME/workspace/ucx/contrib/../test/gtest/common/test.h:19,
                 from $HOME/workspace/ucx/contrib/../test/gtest/ucs/test_pgtable.cc:7:
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_tree.h: In instantiation of ‘static const _Key& std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_S_key(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Link_type) [with _Key = ucs_pgt_region*; _Val = ucs_pgt_region*; _KeyOfValue = std::_Identity<ucs_pgt_region*>; _Compare = test_pgtable_perf::region_comparator; _Alloc = std::allocator<ucs_pgt_region*>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Const_Link_type = const std::_Rb_tree_node<ucs_pgt_region*>*]’:
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_tree.h:2084:47:   required from ‘std::pair<std::_Rb_tree_node_base*, std::_Rb_tree_node_base*> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_get_insert_unique_pos(const key_type&) [with _Key = ucs_pgt_region*; _Val = ucs_pgt_region*; _KeyOfValue = std::_Identity<ucs_pgt_region*>; _Compare = test_pgtable_perf::region_comparator; _Alloc = std::allocator<ucs_pgt_region*>; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::key_type = ucs_pgt_region*]’
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_tree.h:2137:4:   required from ‘std::pair<std::_Rb_tree_iterator<_Val>, bool> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_insert_unique(_Arg&&) [with _Arg = ucs_pgt_region* const&; _Key = ucs_pgt_region*; _Val = ucs_pgt_region*; _KeyOfValue = std::_Identity<ucs_pgt_region*>; _Compare = test_pgtable_perf::region_comparator; _Alloc = std::allocator<ucs_pgt_region*>]’
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_set.h:512:25:   required from ‘std::pair<typename std::_Rb_tree<_Key, _Key, std::_Identity<_Tp>, _Compare, typename __gnu_cxx::__alloc_traits<_Alloc>::rebind<_Key>::other>::const_iterator, bool> std::set<_Key, _Compare, _Alloc>::insert(const value_type&) [with _Key = ucs_pgt_region*; _Compare = test_pgtable_perf::region_comparator; _Alloc = std::allocator<ucs_pgt_region*>; typename std::_Rb_tree<_Key, _Key, std::_Identity<_Tp>, _Compare, typename __gnu_cxx::__alloc_traits<_Alloc>::rebind<_Key>::other>::const_iterator = std::_Rb_tree<ucs_pgt_region*, ucs_pgt_region*, std::_Identity<ucs_pgt_region*>, test_pgtable_perf::region_comparator, std::allocator<ucs_pgt_region*> >::const_iterator; typename __gnu_cxx::__alloc_traits<_Alloc>::rebind<_Key>::other = std::allocator<ucs_pgt_region*>; typename __gnu_cxx::__alloc_traits<_Alloc>::rebind<_Key> = __gnu_cxx::__alloc_traits<std::allocator<ucs_pgt_region*>, ucs_pgt_region*>::rebind<ucs_pgt_region*>; typename _Alloc::value_type = ucs_pgt_region*; std::set<_Key, _Compare, _Alloc>::value_type = ucs_pgt_region*]’
$HOME/workspace/ucx/contrib/../test/gtest/ucs/test_pgtable.cc:414:32:   required from here
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_tree.h:785:15: error: static assertion failed: comparison object must be invocable as const
  785 |               is_invocable_v<const _Compare&, const _Key&, const _Key&>,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
$HOME/workspace/tools/gcc/build-x86_64/install/include/c++/11.0.0/bits/stl_tree.h:785:15: note: ‘std::is_invocable_v<const test_pgtable_perf::region_comparator&, ucs_pgt_region* const&, ucs_pgt_region* const&>’ evaluates to false
```